src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<T>(null, value, null);
+            tail = head;
+
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> nodeToInsert = new Node<>(null, value, null);
+        if (index == 0) {
+            nodeToInsert.next = head;
+            head.prev = nodeToInsert;
+            head = nodeToInsert;
+            size++;
+            return;
+        }
+        Node<T> current = getNodeByIndex(index);
+        nodeToInsert.prev = current.prev;
+        nodeToInsert.next = current;
+        current.prev.next = nodeToInsert;
+        current.prev = nodeToInsert;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
+        }
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
```

Comment:
you don't need to check the index for validity because you do it in the getNodeByIndex() method